plugins/processors/t128_filter/t128_filter.go

```diff
@@ -0,0 +1,112 @@
+package t128_filter
+
+import (
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/processors"
+)
+
+const sampleConfig = `
+[[processors.t128_filter]]
+  ## The conditions that must be met to pass a metric through. This is similar
+  ## behavior to a tagpass, but the multiple tags are ANDed
+  [processors.t128_filter.conditions]
+     #tag1 = ["value1", "value2"]
+	 #tag2 = ["value3"]
+`
+
+type conditionSet map[string][]string
+
+type T128Filter struct {
+	Conditions []conditionSet `toml:"conditions"`
+
+	Log telegraf.Logger `toml:"-"`
+}
+
+func (r *T128Filter) SampleConfig() string {
+	return sampleConfig
+}
+
+func (r *T128Filter) Description() string {
+	return "Filter metrics from being emitted."
+}
+
+func (r *T128Filter) Apply(in ...telegraf.Metric) []telegraf.Metric {
+	filteredPoints := make([]telegraf.Metric, 0)
+
+	for _, point := range in {
+		if doAllConditionSetsMatch(r.Conditions, point) {
+			filteredPoints = append(filteredPoints, point)
+		}
+	}
+
+	return filteredPoints
+}
+
+func doAllConditionSetsMatch(conditionSets []conditionSet, point telegraf.Metric) bool {
+	conditionSetMatches := make([]bool, len(conditionSets))
+	for i, conditionSet := range conditionSets {
+		conditionSetMatches[i] = doesConditionSetMatch(conditionSet, point)
+	}
+	return and(conditionSetMatches)
+}
+
+func doesConditionSetMatch(conditionSet conditionSet, point telegraf.Metric) bool {
+	keyMatches := make([]bool, len(conditionSet))
+	i := 0
+	for key, acceptableValues := range conditionSet {
+		keyMatches[i] = doesKeyMatch(key, acceptableValues, point)
+		i++
+	}
+
+	return and(keyMatches)
+}
+
+func doesKeyMatch(key string, acceptableValues []string, point telegraf.Metric) bool {
+	actualValue, wasSet := point.GetTag(key)
+
+	return wasSet && or(applyBooleanFunc(acceptableValues, func(acceptableValue string) bool {
+		return acceptableValue == actualValue
+	}))
+}
+
+func applyBooleanFunc(items []string, application func(string) bool) []bool {
+	bools := make([]bool, len(items))
+	for i, item := range items {
+		bools[i] = application(item)
+	}
+	return bools
+}
+
+func and(bools []bool) bool {
```

Comment:
This all seems really extra complicated, but it will make things easier in the future when you can configure which func you want to apply to a set of stuff

Reply:
@BenMatase do you have a document/proposal for this that might help understand some of the components. We discussed it at a high level but I'm sure you've put in more thought into it since then.

Reply:
Sure. I'll put one together and send it out